### PR TITLE
Fixed the bug (I hope)

### LINKED ISSUE #16: Configuration starts in completed position
To repeat:
1. Check out Construction-Only branch
2. Start server
3. Open two browsers and navigate to `game.html`
4. Start game and examine configuration

### PR DESCRIPTION
**Pull request by jcolemang**
No longer fixes the goal configuration at whatever the user's current configuration is. I did a little bit of refactoring to maintain lists of block objects instead of parallel lists of block states. I didn't refactor a lot of existing code, but it is a start.

Please actually run it to make sure it works before accepting the changes. This will need a little bit of work to make the changes compatible with the changes in #15 

Fixes #16

### CODE REVIEW

File: 2D Simulation/index.js [hunk 1]
```diff
@@ -78,22 +79,52 @@ var rainbow_select = 0;
...
-}
-
 function getBlockStyle(blockColors) {
     let doc = '';
     for (var i = 0; i < NumBlocks ;i++) {
-        doc += "<style> #block" + i + " {width: 3.8%; height: 7.35%; background-color:"+blockColors[i] +"; border:#000 solid 4px; cursor: move; position: absolute; z-index: 1; text-align: center; vertical-align: middle; line-height: 7.35%; font-family: 'Corben', Georgia, Times, serif;} </style>";
+        doc += "<style> #block" + i +
+            " {width: 3.8%; height: 7.35%; background-color:" +
+            blockColors[i] +
+            "; border:#000 solid 4px; cursor: move; position: absolute; z-index: 1; text-align: center; vertical-align: middle; line-height: 7.35%; font-family: 'Corben', Georgia, Times, serif;} </style>";
```
Comment (jcolemang) [inline 2D Simulation/index.js:88]: This was just a reformat I left in the commit

File: 2D Simulation/initial_board.html [hunk 2]
```diff
@@ -28,13 +28,25 @@
         var container = document.createElement('div');
 
         container.innerHTML = localStorage.getItem("container");
+
+
+     console.log(container.innerHTML);
```
Comment (Lewis-Kelley) [inline 2D Simulation/initial_board.html:33]: Should probably remove this.

**Review by Lewis-Kelley — APPROVED** [review]
It works on my computer.

Looks good to me. There's a few `console.log` entries you left in there you might want to take out.

If you're concerned about the changes in here conflicting with those in the `python-backend` branch, I can just checkout earlier versions of the relevant files to effectively undo what little refactoring I did.

File: 2D Simulation/index.js [hunk 3]
```diff
@@ -417,7 +448,13 @@ function centroid(x, y) {
     return center;
 }
 
-function scoreCal() {
+function scoreCal(finalBlocks) {
+    let goal_left = finalBlocks.map(block => block.position[0]);
+    let goal_top = finalBlocks.map(block => block.position[1]);
+
+    console.log(goal_left);
```
Comment (Lewis-Kelley) [inline 2D Simulation/index.js:455]: Should probably remove these debug statements.

File: 2D Simulation/initial_board.html [hunk 4]
```diff
@@ -28,13 +28,25 @@
...
         container.style.fontSize = "15px";
         container.style.fontWeight = 'bold';
 
         container.id = 'copy_container';
 
         document.body.appendChild(container);
-        // $('#copy_container').addClass('shrink');
+
+        let finalBlocks = JSON.parse(localStorage.getItem('finalBlocks'));
+        console.log(finalBlocks);
```
Comment (Lewis-Kelley) [inline 2D Simulation/initial_board.html:42]: And this.

Comment (jcolemang) [reply 2D Simulation/initial_board.html:42]: I will need to make a bunch of changes to make all of this compatible with your file moves, so I can take these out then.

Comment (jcolemang) [reply 2D Simulation/initial_board.html:42]: Whoops just saw your comment above. The js definitely should not be in the html files so I am all about your changes, I will just change the base branch later later today and actually remove the logs.

**Comment by jcolemang** [comment]
I tried to do git sorcery but I don't think that git can actually do magic. I resolved the conflicts in the UI, I hope that didn't cause any issues.

**Review by kgroble — APPROVED** [review]

**Pull request merged**